pkg/scheduler/scheduler.go

```diff
@@ -136,24 +138,32 @@ func (pc *Scheduler) loadSchedulerConf() {
 	klog.V(4).Infof("Start loadSchedulerConf ...")
 	defer func() {
 		actions, plugins := pc.getSchedulerConf()
-		klog.V(2).Infof("Successfully loaded Scheduler conf, actions: %v, plugins: %v", actions, plugins)
+		klog.V(2).Infof("Finished loading scheduler config. Final state: actions=%v, plugins=%v", actions, plugins)
 	}()
 
+	if pc.disableDefaultConf && len(pc.schedulerConf) == 0 {
+		klog.Fatalf("No --scheduler-conf path provided and default configuration fallback is disabled")
+	}
+
 	var err error
-	pc.once.Do(func() {
-		pc.actions, pc.plugins, pc.configurations, pc.metricsConf, err = UnmarshalSchedulerConf(DefaultSchedulerConf)
-		if err != nil {
-			klog.Errorf("unmarshal Scheduler config %s failed: %v", DefaultSchedulerConf, err)
-			panic("invalid default configuration")
-		}
-	})
+	if !pc.disableDefaultConf {
+		pc.once.Do(func() {
+			pc.actions, pc.plugins, pc.configurations, pc.metricsConf, err = UnmarshalSchedulerConf(DefaultSchedulerConf)
+			if err != nil {
+				klog.Fatalf("Invalid default configuration: unmarshal Scheduler config %s failed: %v", DefaultSchedulerConf, err)
+			}
+		})
+	}
 
 	var config string
 	if len(pc.schedulerConf) != 0 {
 		confData, err := os.ReadFile(pc.schedulerConf)
 		if err != nil {
 			klog.Errorf("Failed to read the Scheduler config in '%s', using previous configuration: %v",
 				pc.schedulerConf, err)
+			if pc.disableDefaultConf {
```

Comment:
I think the judgement of disableDefaultConf and klog.Fatalf should put before the klog.Errorf